utils/service_discovery/sd_docker_backend.py

```diff
@@ -314,6 +314,10 @@ def get_tags(self, state, c_id):
             namespace = pod_metadata.get('namespace')
             tags.append('kube_namespace:%s' % namespace)
 
+            # get kubernetes container name
+            kube_container_name = state.get_kube_container_name(c_id)
```

Comment:
Although probably not plausible, `get_kube_container_name` could return `None` as we get() in a dict. You should add a `if kube_container_name` before appening the tag